samza-test/src/test/java/org/apache/samza/test/framework/StreamApplicationIntegrationTestHarness.java

```diff
@@ -128,32 +129,32 @@ public int clusterSize() {
     return Integer.parseInt(KafkaConfig.TOPIC_DEFAULT_REPLICATION_FACTOR());
   }
 
-
   /**
    * Read messages from the provided list of topics until {@param threshold} messages have been read or until
    * {@link #numEmptyPolls} polls return no messages.
    *
    * The default poll time out is determined by {@link #POLL_TIMEOUT_MS} and the number of empty polls are
    * determined by {@link #numEmptyPolls}
    *
-   * @param topics the list of topics to consume from
+   * @param topic the topic to consume from
    * @param threshold the number of messages to consume
    * @return the list of {@link ConsumerRecord}s whose size can be atmost {@param threshold}
    */
-  public List<ConsumerRecord<String, String>> consumeMessages(Collection<String> topics, int threshold) {
+  public List<ConsumerRecord<String, String>> consumeMessages(String topic, int threshold) {
     int emptyPollCount = 0;
     List<ConsumerRecord<String, String>> recordList = new ArrayList<>();
-    consumer.subscribe(topics);
```

Comment:
Sorry should have been more clear, I was talking about the `consumer` here which is from `IntegrationTestHarness` which this class extends, which has multiple uses except for this one.

Do we need that `consumer` to be `kafkaConsumer` map instead?
I wanted to verify that this class does not have any conflict with the base class when using different consumers

Reply:
In base class IntegrationTestHarness, the only usage of the consumer is in setUp() and tearDown(), which create and close the consumer. For tests that extend StreamApplicationIntegrationTestHarness the only usage of the consumer is here. This means if we don't use the consumer here, there would be no other places for them to access the consumer.

I don't think it is good to use the kafkaConsumer map for the base class since it could be extended by other classes and they don't have this consumeMessage() method that breaks tests. I prefer to only apply the fix to where it is needed

Reply:
If the only use is setup or teardown, lets make sure we follow the lifecycle by:
1. Adding the consumer construction in setup() instead of in consume() and 
2. Adding consumer.stop() for each (currently missing)

Reply:
1. we can't create all consumers in setup because we don't know the topics in advance. added a comment to note that the topicConsumerMap will not be adding all consumers in setup()
2. yes, added a tearDown() to unsubscribe and close all consumers